hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/util/IMetaStoreClientUtil.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.hive.util;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+
+import java.lang.reflect.InvocationTargetException;
+
+public class IMetaStoreClientUtil {
+  public static IMetaStoreClient getMSC(HiveConf hiveConf) throws HiveException, MetaException {
+    IMetaStoreClient metaStoreClient;
+    try {
+      metaStoreClient = ((Hive) Hive.class.getMethod("getWithoutRegisterFns", HiveConf.class).invoke(null, hiveConf)).getMSC();
```

Comment:
Can we give some docs about why this is needed?

Reply:
It can avoid initializing all functions in the environment where Hudi and Spark are integrated. This piece comes from the [PR](https://github.com/apache/hudi/pull/738).

Reply:
Okay, seems a pure code refactoring, my suggestion is just to reuse the tool method, there is no need to change the constructor of these executor classes.

Reply:
In addition to refactoring, another purpose is to pass the `IMetaStoreClient` in `HoodieHiveSyncClient` to these executor classes, so it is necessary to change the constructor of these executor classes.